go-controller/pkg/ovn/egressip.go

```diff
@@ -772,8 +772,32 @@ func (oc *Controller) executeCloudPrivateIPConfigChange(egressIPName string, toA
 			}
 		}
 	}
+	// Merge ops into the existing pendingCloudPrivateIPConfigsOps.
+	// This allows us to:
+	// a) execute only the new ops
+	// b) keep track of any pending changes
 	if len(ops) > 0 {
-		oc.eIPC.pendingCloudPrivateIPConfigsOps[egressIPName] = ops
+		if _, ok := oc.eIPC.pendingCloudPrivateIPConfigsOps[egressIPName]; !ok {
+			// Set all operations for the EgressIP object if none are in the cache currently.
+			oc.eIPC.pendingCloudPrivateIPConfigsOps[egressIPName] = ops
+		} else {
+			for cloudPrivateIP, op := range ops {
+				if _, ok := oc.eIPC.pendingCloudPrivateIPConfigsOps[egressIPName][cloudPrivateIP]; !ok {
+					// If this specific EgressIP object's CloudPrivateIPConfig address currently has no
+					// op, simply set it.
+					oc.eIPC.pendingCloudPrivateIPConfigsOps[egressIPName][cloudPrivateIP] = op
+				} else if op.toAdd != "" {
```

Comment:
Hm... I may be missing something but, what if the toDelete in `op` is not nil?

Reply:
If the existing op has a `toDelete` then that would be the most relevant one and we should not need to overwrite.
If the existing op does not have a `toDelete`, then it would not make sense for the new op to have a `toDelete` either, but perhaps it would be a safe approach to set it if it is possible that missing a number of events can lead to that situation.
WDYT @andreaskaris ?

Reply:
@flavio-fernandes @jcaamano I updated it (and yes, I could have used a continue there,  but given the number of changes I've already made and the fact that I have to cherry-pick downstream to test, I opted for the solution that seemed least error prone to me ;-)